pkg/pmem-csi-driver/main.go

```diff
@@ -71,7 +67,7 @@ func Main() int {
 	klog.V(3).Info("Version: ", version)
 
 	if config.schedulerListen != "" {
-		if config.Mode != Controller {
+		if config.Mode != Webhooks {
```

Comment:
I feel like offloading the webhook and scheduler extension part from the CSI driver to a separate binary makes much sense and cleaner than treating it as a mode of the driver.

Reply:
What's the benefit? Have you considered the drawbacks?

"Cleaner" is subjective.

Reply:
Is this perhaps about publishing one image that has multiple purposed vs. two different images with different content and different purpose?

I prefer to stick with a single image until someone has some strong argument against it. I doubt that the size of an image that only runs once in a cluster matters that much and publishing a single image is simpler, so I prefer that.

When we have a single image, then a single binary is likely to be smaller overall.

What that binary is called and how it is invoked hardly matters for end-users - they never invoke the binary. Renaming the binary is possible, but makes this PR more complex for (IMHO) little gain.

Reply:
My point was not about different images, more about a different binary. As now there is no central controller  driver component, IMHO the scheduler code is not suitable to be part of the driver code.

Reply:
We discussed this in the standup meeting.

"pmem-csi-driver" has already before served multiple different purposes and we agreed to not change that in this PR because doing so would make our image larger for not much gain (basically cleaner command line arguments of the binary which isn't that important).

We also raised the question whether "pmem-csi-operator" should be a separate binary. Probably not. But again, that is a different issue.